src/test/groovy/nebula/test/dependencies/GradleDependencyGeneratorSpec.groovy

```diff
@@ -121,6 +189,26 @@ class GradleDependencyGeneratorSpec extends Specification {
         pom.contains '<scope>compile</scope>'
     }
 
+    def 'check maven pom with dependency with classifier and extension'() {
+        def directory = 'build/testdependencies/mavenpom'
+        def graph = ['test.maven:foo:1.0.0 -> test.maven:bar:1.+:baz@zip']
+        def generator = new GradleDependencyGenerator(new DependencyGraph(graph), directory)
+
+        when:
+        generator.generateTestMavenRepo()
+
+        then:
+        def repo = new File(directory)
+        new File(repo, 'test.maven.foo_1_0_0/build.gradle').text.contains 'compile \'test.maven:bar:1.+:baz@zip\''
+        def pom = new File(repo, 'mavenrepo/test/maven/foo/1.0.0/foo-1.0.0.pom').text
+        pom.contains '<groupId>test.maven</groupId>'
```

Comment:
almost seems like you want to test that the entire block exists. Once you strip whitespace/newlines you should be able to test for contains